[release][CI] stress_test_state_api_scale

### What happened + What you expected to happen

Failed nightly test - https://buildkite.com/ray-project/release-tests-branch/builds/1090#01839fe4-55aa-4dee-86db-732e1d25bb9a

Cluster - https://console.anyscale-staging.com/o/anyscale-internal/projects/prj_qC3ZfndQWYYjx2cz8KWGNUL4/clusters/ses_CJfAadt1FRvztvkkRVS75b7k?command-history-section=command_history

```
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/state_api_test_utils.py", line 79, in invoke_state_api
    assert verify_cb(res), f"Calling State API failed. len(res)=({len(res)}): {res}"
AssertionError: Calling State API failed. len(res)=(0): []
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/state_api_test_utils.py", line 79, in invoke_state_api
    assert verify_cb(res), f"Calling State API failed. len(res)=({len(res)}): {res}"
AssertionError: Calling State API failed. len(res)=(0): []

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "stress_tests/test_state_api_scale.py", line 423, in <module>
    test()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "stress_tests/test_state_api_scale.py", line 369, in test
    test_many_tasks(num_tasks=n)
  File "stress_tests/test_state_api_scale.py", line 92, in test_many_tasks
    limit=STATE_LIST_LIMIT,
  File "stress_tests/test_state_api_scale.py", line 49, in invoke_state_api_n
    invoke_state_api(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/state_api_test_utils.py", line 84, in invoke_state_api
    ), f"Calling {state_api_fn.__name__}({kwargs}) failed with {repr(e)}."
AssertionError: Calling list_tasks({'filters': [('name', '=', 'pi4_sample()')], 'limit': 1000000, 'timeout': 600, 'raise_on_missing_output': False}) failed with AssertionError('Calling State API failed. len(res)=(0): []').
```

### Versions / Dependencies

Master

### Reproduction script

As above

### Issue Severity

High: It blocks me from completing my task.

--- a/release/nightly_tests/stress_tests/test_state_api_scale.py
+++ b/release/nightly_tests/stress_tests/test_state_api_scale.py
@@ -57,7 +57,7 @@ def test_many_tasks(num_tasks: int):
     invoke_state_api(
         lambda res: len(res) == 0,
         list_tasks,
-        filters=[("name", "=", "pi4_sample()"), ("scheduling_state", "=", "RUNNING")],
+        filters=[("name", "=", "pi4_sample"), ("scheduling_state", "=", "RUNNING")],
         key_suffix="0",
         limit=STATE_LIST_LIMIT,
     )
@@ -87,7 +87,7 @@ def test_many_tasks(num_tasks: int):
     invoke_state_api_n(
         lambda res: len(res) == num_tasks,
         list_tasks,
-        filters=[("name", "=", "pi4_sample()")],
+        filters=[("name", "=", "pi4_sample")],
         key_suffix=f"{num_tasks}",
         limit=STATE_LIST_LIMIT,
     )
@@ -101,7 +101,7 @@ def test_many_tasks(num_tasks: int):
     invoke_state_api(
         lambda res: len(res) == 0,
         list_tasks,
-        filters=[("name", "=", "pi4_sample()"), ("scheduling_state", "=", "RUNNING")],
+        filters=[("name", "=", "pi4_sample"), ("scheduling_state", "=", "RUNNING")],
         key_suffix="0",
         limit=STATE_LIST_LIMIT,
     )

